src/velocity.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy.interpolate import interp2d
+
+def interpolated_value(lat, long, vel_array, xy_positions, step = 0.01):
+    """Returns the interpolated value of the tensor for the latitude and longitude requested.
+    
+    """
+    
+    idx = (np.abs(xy_positions[0] - lat)).argmin()
+    idy = (np.abs(xy_positions[1] - long)).argmin()
+    
+    
+    if xy_positions[0][idx] < lat: idx += 1
+    if xy_positions[1][idy] < long: idy += 1
+        
+    x = xy_positions[0][idx:idx+2]
+    y = xy_positions[1][idy:idy+2]
+    
+    small_array = vel_array[idx:idx+2,idy:idy+2,:]
+
+    velx = interp2d(x, y, small_array[:,:,0])
+    
+    vely = interp2d(x, y, small_array[:,:,1])
+    
+    new_velx = velx(lat, long)
+    new_vely = vely(lat, long)
+    
+    #Interpolated square
+    lat_s = np.arange(xy_positions[0][idx], xy_positions[0][idx+1], step)
+    long_s = np.arange(xy_positions[1][idx], xy_positions[1][idx+1], step)
+    velx_s = velx(lat_s, long_s)
+    vely_s = vely(lat_s, long_s)
+        
+    return new_velx, new_vely, lat_s, long_s, velx_s, vely_s
+    
+    
+    
+    
+def velocity_composition(boat_velocity, boat_position, stream_velocitites, xy_stream_velocities):
+    
+    """Given the boat velocity and the stream velocity, computes the total velocity of the boat.
+    """
```

Comment:
Añado explicación de cada parámetro. Corrígeme si alguna explicación no es correcta.

```suggestion
    """
    Given the boat velocity and the stream velocity, computes the total velocity of the boat.
    
    Params
    --------
    boat_velocity : tuple of numpy.array
        The equivalent velocity of the boat if there were not streams.
        boat_velocity is a tuple for both the angle (theta, in radians)
        and the velocity_mod (in the same units as the stream_velocities)
    boat_position : tuple
        Latitude and longitude where the boat is located at the moment.
    stream_velocitites : numpy.array
        The velocity (v_x, v_y) of the stream at each point.
        shape = (lat, lon, 2)
    xy_stream_velocities :  tuple of numpy.array
       Positions of each stream. For instance,
       stream_velocitites[i, j, :] corresponds to position
       ( xy_stream_velocities[0][i], xy_stream_velocities[1][j] )
    Returns
    ---------
    new_vel : tuple
       The composition between both boat and stream velocities.
    """
```